emscripten.py

```diff
@@ -448,6 +452,15 @@ def get_metadata(infile, outfile, modify_wasm, args):
   return metadata
 
 
+def create_sourcemapping_url_file(infile, url):
+  'Create a file with the contents of the sourceMappingURL section'
+  filename = os.path.join(shared.get_emscripten_temp_dir(), infile + '.smu')
```

Comment:
I think you can just do `shared.get_temp_files().get_file('.smu')`

Reply:
It looks the URL file is supposed to exist after this file returns, but with `get_file`, is that possible? All uses of that methods tend to use the idiom of
```py
with shared.get_temp_files().get_file('.ext') as filename:
  do something
```
and the file is gone after this `with` block.

Reply:
Ah, in that case maybe `shared.get_temp_files().get(...)` is what you want then.

Reply:
Done

Reply:
Hmm, come to think of it, the original code assumes a specific filename, not a generated random name. We can probably use `shared.get_temp_files().get_dir()`, but I guess it isn't very different from what @dschuff wrote. I'll revert it.

Reply:
I still think doesn't look right.  Why can't `shared.get_temp_files().get(...)` work?   I don't think the calling code should care what the filename is. 

Also, how about calling this `.sec` (for section?)?  Or even `.bin`?

Reply:
I think it can work if we just delete `create_sourcemapping_url_file` and inline its contents. That way we can use `with` clause and don't need to delete the file afterwards. Currently we just leave the temporary file. I also changed its extension to `.bin` as suggested. So I deleted `create_sourcemapping_url_file` and changed its callsite to:
```py
    else:
      # Otherwise use objcopy. This avoids re-encoding the file (thus
      # preserving DWARF) and is faster.

      # Create a file with the contents of the sourceMappingURL section    
      with shared.get_temp_files().get_file('.bin') as url_file:
        utils.write_binary(url_file,
                           leb128.u.encode(len(base_url)) + base_url.encode('utf-8'))
        cmd = [shared.LLVM_OBJCOPY,
               '--add-section',
               'sourceMappingURL=' + url_file,
               infile]
        shared.check_call(cmd)
```